### PR TITLE
fix: Make new Endpoint to delete favourite using EventId

### LINKED ISSUE #5857: Delete User Favorite By EventId
**Is your feature request related to a problem? Please describe.**
Current Implementation of user-favorite-events allows deletion of events by using the id given by the server.
**Describe the solution you'd like**
There should be an option to delete the event using eventId rather the user-favorite-event id

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #5857 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Now a user can delete the event being favorite using Event ID.

### CODE REVIEW

File: app/api/user_favourite_events.py [hunk 1]
```diff
@@ -94,3 +98,25 @@ class UserFavouriteEventRelationship(ResourceRelationship):
...
+        id = 'id'
+
+    event = safe_query(db, Event, id, identifier, 'event_' + id)
+
+    user_favourite_event = UserFavouriteEvent.query.filter_by(user=current_user, event_id=event.id).first()
+    if user_favourite_event:
+        db.session.delete(user_favourite_event)
+        db.session.commit()
+    else:
+        raise ConflictException({'pointer': '/data/relationships/event'}, "Event is not favourited")
```
Comment (iamareebjamal) [inline app/api/user_favourite_events.py]: Why ConflictException?

Comment (prateekj117) [reply app/api/user_favourite_events.py]: @iamareebjamal Corrected it. Please review.

**Comment by iamareebjamal** [comment]
Please write here the endpoint and params required

**Comment by prateekj117** [comment]
@iamareebjamal 
Screenshot from 2019-06-01 08-01-21

**Comment by prateekj117** [comment]
@iamareebjamal Please let me know what other things are required in this PR.

**Comment by iamareebjamal** [comment]
To get a PR reviewed, remove WIP from the title. Nobody reviews WIP PRs. WIP is explicit indication that the PR is incomplete and not ready for review

**Comment by prateekj117** [comment]
@iamareebjamal Any changes in this?

**Comment by iamareebjamal** [comment]
Will see today

**Comment by prateekj117** [comment]
@iamareebjamal Here, I will have to rename the `id` of Events Model to `event_id`, that would not be a trouble for any other functionality, right?

**Comment by iamareebjamal** [comment]
@mrsaicharan1 @uds5501 @shreyanshdwivedi Are we using it in frontend?

**Comment by uds5501** [comment]
@iamareebjamal No.

**Comment by prateekj117** [comment]
@iamareebjamal So should I rename it?

**Comment by shreyanshdwivedi** [comment]
@prateekj117 can you please explain briefly why do you need to  changes id to `event_id` in event model?

**Comment by iamareebjamal** [comment]
@shreyanshdwivedi Read the issue. Also, the docs specify that the id is actually event ID and not event-favorite ID

**Comment by prateekj117** [comment]
@iamareebjamal Please review. Tested. It's working.

**Comment by iamareebjamal** [comment]
Update api docs to pass the build

**Comment by iamareebjamal** [comment]
This will delete all the event favorite made by any user. Please write a query to delete only user's favorite

**Comment by prateekj117** [comment]
@iamareebjamal Please review.

**Comment by iamareebjamal** [comment]
Still wrong. Please test before sending. Now even the logic of permission is wrong.

**Comment by prateekj117** [comment]
@uds5501 @mrsaicharan1 @iamareebjamal The routes work properly locally, but the builds are failing, any idea why?

**Comment by iamareebjamal** [comment]
Yes because event ID in delete docs is route

File: docs/api/api_blueprint.apib [hunk 2]
```diff
@@ -25641,9 +25641,9 @@ This Group's APIs can be used for adding a particular event to the favourite lis
             }
         }
 
-## Delete Favourite Events [/v1/user-favourite-events/{event_id}]
+## Delete Favourite Events [/v1/user-favourite-events/{event_identifier}]
```
Comment (iamareebjamal) [inline docs/api/api_blueprint.apib]: Should be event_id identifier

File: docs/api/api_blueprint.apib [hunk 3]
```diff
@@ -25641,9 +25641,9 @@ This Group's APIs can be used for adding a particular event to the favourite lis
...
 + Parameters
-    + event_id: 1 (integer) - ID of the Event
+    + event_identifier: 1 (string) - identifier or event id of the event. (b8324ae2 is an example of identifier)
```
Comment (iamareebjamal) [inline docs/api/api_blueprint.apib]: Completely wrong. Revert

Comment (prateekj117) [reply docs/api/api_blueprint.apib]: @iamareebjamal Only event_id (without identifier) is not being used anywhere in the API. Can you please help me a bit more regarding this.

Comment (iamareebjamal) [reply docs/api/api_blueprint.apib]: It's being used literally everywhere. Event identifier is only being used in 1 place on frontend

Comment (prateekj117) [reply docs/api/api_blueprint.apib]: @iamareebjamal In API docs, I cannot find a single example where only `event_id` is used. Can you please point me to one.

**Comment by prateekj117** [comment]
@uds5501 @mrsaicharan1 @shreyanshdwivedi Can you please review my last two commits and tell me what am I doing wrong in testing.

**Comment by bhaveshAn** [comment]
@prateekj117 

> Can you please review my last two commits and tell me what am I doing wrong in testing.

You added factory object of '`EventFactory`  instead of `UserFavouriteEventFactory` this creates object of Event to delete object of favourite_event, which makes no sense. Hence add `UserFavouriteEventFactory` object in `hock_main` instead of `EventFactory`
Thanks!

**Comment by prateekj117** [comment]
@bhaveshAn This I did because I am using event_id now instead of id of user_favourite_event. I already tried with the previous hook. Please see my the changes I tried in last 3 commits and let me know what I am doing wrong. Thank you.

**Comment by iamareebjamal** [comment]
He already did. You need to understand how the dredd testing and factory methods work. I know exactly what's wrong, but there'll be no use of learning if I just tell you that.

First try to understand why it's failing. Because right now you're just changing random things and hoping it'll work. Which is not better it being fixed itself.

Hardly takes 30 minutes to 1 hour to understand the working of dredd and factory mechanism. Takes more than heat death of universe to change things randomly and hoping it'll get fixed

**Pull request closed without merging**